src/algebra/ordered_group.lean

```diff
@@ -526,6 +527,16 @@ def function.injective.ordered_comm_group [ordered_comm_group α] {β : Type*}
   ..hf.ordered_comm_monoid f one mul,
   ..hf.comm_group f one mul inv div }
 
+/-- `equiv.mul_left` as an order_iso. -/
+@[simps {simp_rhs := tt}]
+def order_iso.mul_left [ordered_comm_group α] (a : α) : α ≃o α :=
+{ map_rel_iff' := λ _ _, mul_le_mul_iff_left a, ..equiv.mul_left a }
+
+/-- `equiv.mul_right` as an order_iso. -/
+@[simps {simp_rhs := tt}]
+def order_iso.mul_right [ordered_comm_group α] (a : α) : α ≃o α :=
+{ map_rel_iff' := λ _ _, mul_le_mul_iff_right a, ..equiv.mul_right a }
+
```

Comment:
@adomani, it looks like my changes crossed with the order refactor. Where should these now go, and what assumptions should they have?

Reply:
Eric, I am not at a computer right now, but, if they compile where they are, they can stay there.  Since these are specific about `ordered_comm_group` I would not weaken the assumptions.  At a later stage, there may be a choice of removing altogether the `ordered_[...]` typeclasses, but this was not my initial intention: I simply wanted to have the possibility of separating order and operation assumptions.

If you feel like it, you could create a `section ordered_comm_group` and gather these results together, but if it is only these few lemmas it may not be worth it.

Reply:
These aren't really specific to `ordered_comm_group`, they just need `mul_le_mul_iff_left` and a `group` structure. Is there a better way to state that now?

Reply:
Ah, how about trying `group` and `covariant_class α α (*) (≤)`.  Note that I am again without computer...

Reply:
And of course `partial_order` as well!  You might get away with much less than this, tough.  Try with `monoid` and `preorder`, and see if it flies...

Reply:
You might even just want to try to transport `covariant_class α α (*) (≤)` directly.  For that typeclass, assuming `has_mul` and `has_le` is  enough.

Reply:
Great, I think I found a new home for them